upkie/cpp/actuation/BulletInterface.cpp

```diff
@@ -196,8 +197,16 @@ void BulletInterface::observe(Dictionary& observation) const {
       Eigen::Vector3d(T(0, 3), T(1, 3), T(2, 3));  // [m]
   monitor("base")("orientation") =
       Eigen::Quaterniond(T.block<3, 3>(0, 0));  // [w, x, y, z]
-}
 
+  // Observe the environnement urdf states
+  for (int urdf_id : urdf_ids) {
+    Eigen::Matrix4d T = transform_extra_urdf_to_world(urdf_id);
+    monitor("extra_urdf_" + std::to_string(urdf_id))("position") =
```

Comment:
```suggestion
    monitor(body_name)("position") =
```

(where you can get `body_name` and `body_id` by iterating over the map)